modules/aruco/src/apriltag_quad_thresh.cpp

```diff
@@ -0,0 +1,1600 @@
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+The views and conclusions contained in the software and documentation are those
+of the authors and should not be interpreted as representing official policies,
+either expressed or implied, of the Regents of The University of Michigan.
+ */
+
+// limitation: image size must be <32768 in width and height. This is
+// because we use a fixed-point 16 bit integer representation with one
+// fractional bit.
+
+
+#include "apriltag_quad_thresh.hpp"
+
+//#define APRIL_DEBUG
+#ifdef APRIL_DEBUG
+    #include "opencv2/imgcodecs.hpp"
+    #include <opencv2/imgproc.hpp>
+#endif
+
+namespace cv {
+namespace aruco {
+
+static inline void ptsort(struct pt *pts, int sz){
+#define MAYBE_SWAP(arr,apos,bpos)                                   \
+        if (arr[apos].theta > arr[bpos].theta) {                        \
+            tmp = arr[apos]; arr[apos] = arr[bpos]; arr[bpos] = tmp;    \
+        };
+
+    if (sz <= 1)
+        return;
+
+    if (sz == 2) {
+        struct pt tmp;
+        MAYBE_SWAP(pts, 0, 1);
+        return;
+    }
+
+    // NB: Using less-branch-intensive sorting networks here on the
+    // hunch that it's better for performance.
+    if (sz == 3) { // 3 element bubble sort is optimal
+        struct pt tmp;
+        MAYBE_SWAP(pts, 0, 1);
+        MAYBE_SWAP(pts, 1, 2);
+        MAYBE_SWAP(pts, 0, 1);
+        return;
+    }
+
+    if (sz == 4) { // 4 element optimal sorting network.
+        struct pt tmp;
+        MAYBE_SWAP(pts, 0, 1); // sort each half, like a merge sort
+        MAYBE_SWAP(pts, 2, 3);
+        MAYBE_SWAP(pts, 0, 2); // minimum value is now at 0.
+        MAYBE_SWAP(pts, 1, 3); // maximum value is now at end.
+        MAYBE_SWAP(pts, 1, 2); // that only leaves the middle two.
+        return;
+    }
+
+    if (sz == 5) {
+        // this 9-step swap is optimal for a sorting network, but two
+        // steps slower than a generic sort.
+        struct pt tmp;
+        MAYBE_SWAP(pts, 0, 1); // sort each half (3+2), like a merge sort
+        MAYBE_SWAP(pts, 3, 4);
+        MAYBE_SWAP(pts, 1, 2);
+        MAYBE_SWAP(pts, 0, 1);
+        MAYBE_SWAP(pts, 0, 3); // minimum element now at 0
+        MAYBE_SWAP(pts, 2, 4); // maximum element now at end
+        MAYBE_SWAP(pts, 1, 2); // now resort the three elements 1-3.
+        MAYBE_SWAP(pts, 2, 3);
+        MAYBE_SWAP(pts, 1, 2);
+        return;
+    }
+
+#undef MAYBE_SWAP
+
+    // a merge sort with temp storage.
+
+    // Use stack storage if it's not too big.
+    int stacksz = sz;
+    if (stacksz > 1024)
+        stacksz = 0;
+
+    struct pt *_tmp_stack = new struct pt[stacksz];
```

Comment:
Dynamic array size is C99 extension (not supported in C++).
Perhaps you need to use alloca() instead.
Or `cv::AutoBuffer<struct pt, 1024> _tmp_stack(sz);` (but it doesn't not completely equal to variable size arrays - be careful with recursive functions).

**Update**: I see "new" here. Probably you should use `cv::AutoBuffer<struct pt, 128> _tmp_stack(sz)` here without any "stacksz" trick.